[WIP] Update elasticsearch connector to handle up to v6

n-riesco opened the pull request:
TODO
- [x] ~~connect: fails if user didn't provide the port number~~ (moved to #370)
- [ ] connect: do not use `_cat/indices` to validate the connection (see comment https://github.com/plotly/falcon-sql-client/issues/316#issuecomment-364695779)
- [x] ~~works behind Nginx~~ (moved to #370)
- [x] works with ElasticSearch v5
- [ ] works with ElasticSearch v6

Closes #316 
Closes #317

shannonlal commented:
@n-riesco I would be interested in working with you on these fixes.  Do you want to create a separate branch on the falcon-sql-client that we can merge into?  One other todo that I would like to suggest is that we add the ability to define a query for Elastic Search

n-riesco commented:
@shannonlal I'm sorry I haven't moved this branch into falcon's repo yet. But I wanted to lose the freedom to do a forced git push, before I got the Dockerfile right.

This is how far I got:

- I've added tests that use mocked responses from an elasticsearch v2.4 server (to run this tests: `yarn test-unit-elasticsearch-v2`)

- I've added tests that use mocked responses from an elasticsearch v5.6.7 server (to run this tests: `yarn test-unit-elasticsearch-v5`)

- I've created a docker container that runs an elasticsearch server v5.6.7 with enough sample data to run the tests (to build the image: `yarn docker:elasticsearch:build`; and to run it: `yarn docker:elasticsearch:start`).

The **main result** from the work so far is that Falcon's tests pass with either elasticsearch v2.4 or v5.6.7.

@shannonlal Are you still able to reproduce the errors in [this comment](https://github.com/plotly/falcon-sql-client/issues/316#issuecomment-352223105)?

shannonlal commented:
After talking discussion with @n-riesco we agreed to merge this in 💃 .
